### ISSUE TITLE
About conformer encoder

### ISSUE THREAD

**Issue by Lostsy** [speechbrain/speechbrain#800]
Hi, maybe it's a bug report for the conformer encoder. 

I was working on a conformer ASR model several weeks ago. I met some questions that I cannot reproduce the 12 layer conformer model's result in ESPnet. (You know it's not convenient to modify the model in ESPnet... And the training efficiency was low. ) 

I checked many settings and recently I find that there seems to be a small bug for the Conformer Encoder. Following the Conformer Paper [paper](https://arxiv.org/pdf/2005.08100.pdf), the conformer encoder contains two feed-forward modules. And I think these two feed-forward modules are different modules. However, in the Speechbrain code, the ffn_module is defined only once but used twice.  
<img width="404" alt="conformer_encoder" src="https://user-images.githubusercontent.com/32313215/120340375-8efd3880-c328-11eb-8342-bbd18b541e8d.png">
Figure from [paper](https://arxiv.org/pdf/2005.08100.pdf)

 [Link to the speech brain conformer code](https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/lobes/models/transformer/conformer.py#L138)
```
def forward( 
 self, 
 x, 
 src_mask: Optional[torch.Tensor] = None, 
 src_key_padding_mask: Optional[torch.Tensor] = None, 
 ): 
 # ffn module 
 x = x + 0.5 * self.ffn_module(x) 
   # muti-head attention module 
 x = self.norm1(x) 
 output, self_attn = self.Multihead_attn( 
 x, x, x, attn_mask=src_mask, key_padding_mask=src_key_padding_mask, 
 ) 
 x = x + output 
   # convolution module 
 x = x + self.convolution_module(x) 
   # ffb module 
 y = self.norm2(x + 0.5 * self.ffn_module(x)) 
 return y, self_attn 

```
For my understanding, we should define two ffn_module here. Correct me if I'm wrong! Glad to hear from you~

In fact, the current conformer structure works for me in a 4-layer model. But I cannot achieve a better result in a 12-layer model. 

Best,

Yang

**Comment by TParcollet** [comment]
Hi! We currently are working on a fix for conformer. you can have a look at the PR #535 . @popcornell any opinion on that ? I think we checked in the past, but I don't remember.

**Comment by TParcollet** [comment]
@Lostsy did you try with a second ffn module ?

**Comment by Lostsy** [comment]
Hi,

I think I will try the second ffn module in the recent weeks. But before that, I have another experiment to do and I have no free GPU resources. I will give you my conclusion later~

In fact, in my experiment, I used the relative positional encoding from ESPnet. I trained the model on a ~2000h Chinese-Mandarin dataset. The 4-layer conformer model performs well but the 12-layer model was not as good as the one trained with ESPnet. 

I hope this change can help me improve my 12-layer model. 

Best, 
Yang
